src/zenml/zen_server/routers/users_endpoints.py

```diff
@@ -159,185 +168,214 @@ def get_user(
     return zen_store().get_user(user_name_or_id=user_name_or_id)
 
 
-@router.put(
-    "/{user_name_or_id}",
-    response_model=UserResponseModel,
-    responses={401: error_response, 404: error_response, 422: error_response},
-)
-@handle_exceptions
-def update_user(
-    user_name_or_id: Union[str, UUID],
-    user_update: UserUpdateModel,
-    _: AuthContext = Security(authorize, scopes=[PermissionType.WRITE]),
-) -> UserResponseModel:
-    """Updates a specific user.
-
-    Args:
-        user_name_or_id: Name or ID of the user.
-        user_update: the user to use for the update.
-
-    Returns:
-        The updated user.
-    """
-    user = zen_store().get_user(user_name_or_id)
+# When the auth scheme is set to EXTERNAL, users cannot be updated via the
+# API.
+if server_config().auth_scheme != AuthScheme.EXTERNAL:
```

Comment:
Any reason you query this multiple times instead of wrapping all the endpoints in one if condition?

Reply:
There is an order to these endpoints that I don't want to mess up. I'm only disabling the PUTs and POSTs that are no longer valid.